homeassistant/components/simplefin/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for SimpleFIN integration."""
+
+from typing import Any
+
+from simplefin4py import SimpleFin
+from simplefin4py.exceptions import (
+    SimpleFinAuthError,
+    SimpleFinClaimError,
+    SimpleFinInvalidAccountURLError,
+    SimpleFinInvalidClaimTokenError,
+    SimpleFinPaymentRequiredError,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+
+from .const import CONF_ACCESS_URL, DOMAIN, LOGGER
+
+
+class SimpleFinConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the initial setup of a SimpleFIN integration."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Prompt user for SimpleFIN API credentials."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            access_url = user_input[CONF_ACCESS_URL]
+            try:
+                if not access_url.startswith("http"):
+                    # Claim token detected - convert to access url
+                    LOGGER.debug("[Setup Token] - Claiming Access URL")
+                    access_url = await SimpleFin.claim_setup_token(access_url)
+                else:
+                    LOGGER.debug("[Access Url] - 'http' string detected")
+                    # Validate the access URL
+                    LOGGER.debug("[Access Url] - validating access url")
+                    SimpleFin.decode_access_url(access_url)
+
+                LOGGER.debug("[Access Url] - Fetching data")
+                simple_fin = SimpleFin(access_url=access_url)
+                await simple_fin.fetch_data()
+
+                user_input[CONF_ACCESS_URL] = access_url
+
+                return self.async_create_entry(
+                    title="SimpleFIN",
+                    data={CONF_ACCESS_URL: user_input[CONF_ACCESS_URL]},
+                )
```

Comment:
Only keep stuff in the try block that can raise

Reply:
`claim_setup_token` `decode_access_url` and `fetch_data` all throw similar-ish errors

Reply:
I think it was much cleaner when it was separate functions...

Reply:
I'll pull the stuff after fetch data down.

Reply:
you can create an `else:` block with the last stuff in it